### ISSUE TITLE
The Mighty B! Pier 13 Throwdown 

### ISSUE THREAD

**Issue by Stefannag** [ruffle-rs/ruffle#14837]
### Describe the bug

At the end of preloading, error occurs: 

`ERROR core/src/avm2/events.rs:428 Error dispatching event EventObject(EventObject { type: "complete", class: com.sarbakan.sbdk.events::UIEvent, ptr: 0x1d80c38 }) to handler FunctionObject(FunctionObject { ptr: 0x1802960, name: Ok("PreloaderScreen/onProgressbarComplete()") }) : ArgumentError: Error #2024: An object cannot be added as a child of itself.`

The game is working if you don't load the preloader, and load the main.swf directly, as [here](https://www.getgames.io/games/ruffle/?swf=nick/MightyB/mighty-b-pier/main.swf&params=&name=The%20Mighty%20B!%20Pier%2013%20Throwdown).

### Expected behavior

Should load the game, but it is stuck at preloading.

### Content Location

[Ruffle](https://www.getgames.io/games/ruffle/?swf=nick/MightyB/mighty-b-pier/preloader.swf&params=&name=The%20Mighty%20B!%20Pier%2013%20Throwdown)

### Affected platform

Self-hosted version

### Operating system

Windows 11

### Browser

Google Chrome Version 120.0.6099.200 (Official Build) (64-bit)

### Additional information

Couldn't find a working flash page.

**Comment by n0samu** [comment]
SWF URL is https://www.getgames.io/games/swf/nick/MightyB/mighty-b-pier/preloader.swf
It works in Flash Player so it is definitely a Ruffle bug

**Comment by Stefannag** [comment]
> SWF URL is https://www.getgames.io/games/swf/nick/MightyB/mighty-b-pier/preloader.swf It works in Flash Player so it is definitely a Ruffle bug

Indeed, that is the correct swf, but just wanted to point out that the game is working without the preloader.

**Comment by Lord-McSweeney** [comment]
I cannot reproduce this issue with the latest version of Ruffle; @Stefannag can you confirm that it's been fixed?

**Comment by Lord-McSweeney** [comment]
Closing as I can no longer reproduce this issue. Please reopen this issue if you can still reproduce it!
